.github/workflows/ci.yaml

```diff
@@ -1,47 +1,55 @@
 # This config uses industrial_ci (https://github.com/ros-industrial/industrial_ci.git).
 # For troubleshooting, see readme (https://github.com/ros-industrial/industrial_ci/blob/master/README.rst)
 
-name: Build and Test (rolling)
+name: CI
 
 on:
   workflow_dispatch:
   pull_request:
   push:
     branches:
       - main
+      - test/gha-docker
 
 jobs:
-  industrial_ci:
+  default:
     strategy:
+      fail-fast: false
       matrix:
         env:
-          - ROS_DISTRO: rolling
-            ROS_REPO: main
+          - IMAGE: rolling-ci-testing
             CCOV: true
-          - ROS_DISTRO: rolling
-            ROS_REPO: testing
+          - IMAGE: rolling-ci-testing
             IKFAST_TEST: true
-            CLANG_TIDY: true
+            CLANG_TIDY: pedantic
+          - IMAGE: galactic-ci
+          - IMAGE: galactic-ci-testing
     env:
-      UPSTREAM_WORKSPACE: moveit2_rolling.repos
+      CXXFLAGS: "-Wall -Wextra -Wwrite-strings -Wunreachable-code -Wpointer-arith -Wredundant-decls -Wno-deprecated-copy"
+      DOCKER_IMAGE: ghcr.io/ros-planning/moveit2:${{ matrix.env.IMAGE }}
+      UPSTREAM_WORKSPACE: moveit2.repos $(f="moveit2_$(sed 's/-.*$//' <<< "${{ matrix.env.IMAGE }}").repos"; test -r $f && echo $f)
```

Comment:
We can comment out this if CI is happy in all distros, if not that's good, see comment below on .repos files.